frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
I prefer we don't change the behavior of the core code path for a plugin unless really necessary, otherwise it defeats the whole point of separating the code out as a plugin ;)

How about we add a new `onOpenDocument` event and add an event handler in kosync plugin to generate this hash? Or we can go with the `ReaderUI.postInitiCallback` hook approach to add a new `preInitCallback` hook.